### DISCUSSION TITLE
Editing multiple input values in single form with Controller

### DISCUSSION THREAD

**Discussion by kosaki08** [react-hook-form/react-hook-form#9907] [Q&A] [answered]
### Version Number
7.42.1

### Codesandbox/Expo snack
https://codesandbox.io/s/wonderful-lena-473c1l?file=/src/TestForm.tsx

### Steps to reproduce

1. Select a number from select box.
2. The input fields do not switch.
3. Clear the value in the field (validation failed), select another number, and focus on the input field.
4. Validation passed.

### Expected behavior

- The value of each input field is updated according to the change of control (the item corresponding to the selected index number is displayed).
- Validation is also applied to elements corresponding to index numbers that are not selected (buttons are disabled)

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

No response

### Code of Conduct

I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@kosaki08 

There are several issues on your implementation:

1. The select box to select field number to render its input elements is using value from property `code` which starts at 1 but you're using an array of `testContents` where index starts at 0. This leads to wrong test content for each selected number.
2. You are using same input elements to render UI for each selected number but did not pass prop `key` to define uniqueness when re-rendering. Without prop `key`, React will re-use real DOM input elements even that selected number is changed that leads to unexpected re-rendering UI.

Just add prop `key` to make sure React can replace correct DOM element for re-rendering UI of each selected number.

Working demo: https://codesandbox.io/s/quirky-pare-uzwywr?file=/src/TestForm.tsx

**Reply by kosaki08** [reply]
@leapful 
Thank you a lot! You saved me a lot of time.
